quiche/src/lib.rs

```diff
@@ -18940,93 +18977,145 @@ mod tests {
-        config.set_max_send_udp_payload_size(1350);
-        config.set_max_recv_udp_payload_size(1250);
+        config.set_max_send_udp_payload_size(1400);
         config.discover_pmtu(true);
 
-        // Perform initial handshake.
         let mut pipe = testing::Pipe::with_config(&mut config).unwrap();
         assert_eq!(pipe.handshake(), Ok(()));
 
-        let server_addr = testing::Pipe::server_addr();
-        let client_addr = testing::Pipe::client_addr();
-        let pid_1 = pipe
-            .server
-            .paths
-            .path_id_from_addrs(&(server_addr, client_addr))
-            .expect("no such path");
-
-        // Check that PMTU params are configured correctly
-        let pmtu_param = &mut pipe.server.paths.get_mut(pid_1).unwrap().pmtud;
-        assert!(pmtu_param.get_should_probe());
-        assert_eq!(pmtu_param.get_probe_size(), 1350);
-        std::thread::sleep(
-            pipe.server.paths.get_mut(pid_1).unwrap().recovery.rtt() +
-                time::Duration::from_millis(1),
-        );
+        // Get initial probe size
+        let active_path = pipe.client.paths.get_active_mut().unwrap();
+        let initial_probe_size = active_path.pmtud.get_probe_size();
+        assert_eq!(initial_probe_size, 1400);
+
+        // Send first probe
+        let mut out = [0; 4096];
+        while let Ok(_) = pipe.client.send(&mut out) {}
+
+        // Verify probe flag was reset after sending
+        let active_path = pipe.client.paths.get_active_mut().unwrap();
+        assert!(!active_path.pmtud.get_should_probe());
+
+        // Simulate probe loss
+        active_path.pmtud.pmtu_probe_lost();
+
+        // Verify probe flag is re-enabled and size is reduced
+        assert!(active_path.pmtud.get_should_probe());
+        let second_probe_size = active_path.pmtud.get_probe_size();
+        assert!(second_probe_size < initial_probe_size);
+        assert!(second_probe_size > 1200);
+
+        // Send second probe
+        let mut out = [0; 4096];
+        while let Ok(_) = pipe.client.send(&mut out) {}
+
+        // Verify second probe flag was reset
+        let active_path = pipe.client.paths.get_active_mut().unwrap();
+        assert!(!active_path.pmtud.get_should_probe());
 
-        let active_server_path = pipe.server.paths.get_active_mut().unwrap();
-        let pmtu_param = &mut active_server_path.pmtud;
+        // Simulate second probe loss
+        active_path.pmtud.pmtu_probe_lost();
 
-        // PMTU not updated since probe is not ACKed
-        assert_eq!(pmtu_param.get_current_mtu(), 1200);
+        // Verify third probe is even smaller
+        assert!(active_path.pmtud.get_should_probe());
+        let third_probe_size = active_path.pmtud.get_probe_size();
+        assert!(third_probe_size < second_probe_size);
 
-        // Continue searching for PMTU
-        assert!(pmtu_param.get_should_probe());
+        // Verify binary search convergence
```

Comment:
Worth doing these tests?
- First 2 probes fail like in this test, 3rd succeeds.  MTU updated to 1250.
- All probes fail until MTU settles at 1200.